static/js/components/CandidateAnnotationsList.jsx

```diff
@@ -0,0 +1,157 @@
+import React, { useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import PropTypes from "prop-types";
+
+import { makeStyles } from "@material-ui/core/styles";
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import ListItemText from "@material-ui/core/ListItemText";
+import Collapse from "@material-ui/core/Collapse";
+import ExpandLess from "@material-ui/icons/ExpandLess";
+import ExpandMore from "@material-ui/icons/ExpandMore";
+import Divider from "@material-ui/core/Divider";
+
+import * as candidatesActions from "../ducks/candidates";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: "100%",
+    background: theme.palette.background.paper,
+    padding: theme.spacing(1),
+    maxHeight: "15rem",
+    overflowY: "scroll",
+    // Prevent disabled annotations from being selected, but display normally for readability
+    "& .Mui-disabled": {
+      opacity: 1,
+    },
+  },
+  nested: {
+    paddingLeft: theme.spacing(4),
+    paddingTop: 0,
+    paddingBottom: 0,
+  },
+}));
+
+export const getAnnotationValueString = (value) => {
+  let valueString;
+  const valueType = typeof value;
+  switch (valueType) {
+    case "number":
+      valueString = value.toFixed(4);
+      break;
+    case "string":
+      valueString = `${value.substring(0, 20)}...`;
+      break;
+    case "object":
+      valueString = `${JSON.stringify(value).substring(0, 20)}...`;
+      break;
+    default:
+      valueString = value.toString();
+  }
+  return valueString;
+};
+
+// This component will be used to display annotations in the scanning page.
+// Included in this PR only for the getAnnotationValueString() function above.
+// Can ignore the rest of this component during review of this PR for now.
+const CandidateAnnotationsList = ({ annotations }) => {
+  const classes = useStyles();
+
+  const dispatch = useDispatch();
+
+  const initState = {};
+  annotations.forEach((annotation) => {
+    initState[annotation.origin] = true;
+  });
+  const [openedOrigins, setopenedOrigins] = useState(initState);
+
+  const selectedAnnotationItem = useSelector(
+    (state) => state.candidates.selectedAnnotationItem
+  );
+
+  const handleClick = (origin) => {
+    setopenedOrigins({ ...openedOrigins, [origin]: !openedOrigins[origin] });
+  };
+
+  const handleItemSelect = (origin, key) => {
+    const currentlySelected =
+      selectedAnnotationItem &&
+      selectedAnnotationItem.origin === origin &&
+      selectedAnnotationItem.key === key;
+
+    const annotationItem = currentlySelected ? null : { origin, key };
+    dispatch(candidatesActions.setCandidatesAnnotationItem(annotationItem));
+  };
+
+  return (
+    <List
+      component="nav"
+      aria-labelledby="nested-list-subheader"
+      className={classes.root}
+      dense
+    >
+      {annotations.map((annotation) => (
+        <div key={`annotation_${annotation.origin}`}>
```

Comment:
Isn't it possible for many annotations to have the same origin, or are all annotations with the same origin being combined somewhere?

Reply:
Annotations can have the same origin across candidates, but not within the same candidate, Since this list is used to display the annotations of one specific candidate, there is no issue here.